### PR TITLE
Feature/download pdf

### PR DESCRIPTION
**Pull request by sbangar07**
PDF Download functionality with websocket implementation

### CODE REVIEW

File: cms/config/middlewares.ts [hunk 1]
```diff
@@ -24,9 +36,9 @@ export default [
             formLimit: '56kb',
             jsonLimit: '1mb',
             formidable: {
-                maxFileSize: 200 * 1024 * 1024,
+                maxFileSize: 200 * 1024 * 1024, // 200 MB limit
```
Comment (Nileshdcool) [inline cms/config/middlewares.ts:39]: move this into seperate constant file where add a variable and use variable for the same

File: cms/src/api/pdf/controllers/pdf.ts [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+  async generatePdf(ctx) {
+    try {
+      const userId = Array.isArray(ctx.request.headers['x-user-email'])
+        ? ctx.request.headers['x-user-email'][0]
+        : ctx.request.headers['x-user-email'];
+
+      let selectedQuestions = ctx.request.body?.selectedQuestions || [];
+      const fileName = ctx.request.body?.fileName;
+      const topicName = ctx.request.body?.topicName;
+      const pdfStoragePath = process.env.PDF_STORAGE_PATH || "C:/sbangar/personal/FilePath";
```
Comment (Nileshdcool) [inline cms/src/api/pdf/controllers/pdf.ts:28]: Remove hardcoded file path, it must be dynamic path , and should be created within same directpry on server

File: interfaces/interviewmodels.ts [hunk 3]
```diff
@@ -44,5 +44,10 @@ export interface Topic {
   export interface SectionContentProps {
     selectedSection: number;
     filteredQaList: Question[];
+    topicName:string ;
   }
 
+  export interface WebSocketData {
+    message: string; // Type of the progress message
+    status: 'completed' | 'in-progress' | 'failed'; // Example statuses
```
Comment (Nileshdcool) [inline interfaces/interviewmodels.ts:52]: Create Enum and use enum variable here

**Review by Nileshdcool — CHANGES_REQUESTED** [review]

File: cms/config/middlewares.ts (hunk 1 above)
Comment (sbangar07) [reply cms/config/middlewares.ts]: Resolved

File: cms/src/api/pdf/controllers/pdf.ts (hunk 2 above)
Comment (sbangar07) [reply cms/src/api/pdf/controllers/pdf.ts]: Resolved

File: interfaces/interviewmodels.ts (hunk 3 above)
Comment (sbangar07) [reply interfaces/interviewmodels.ts]: Resolved

**Pull request merged**